ambry-quota/src/main/java/com/github/ambry/quota/PreProcessQuotaChargeCallback.java

```diff
@@ -54,7 +52,12 @@ public QuotaAction checkAndCharge(boolean shouldCheckQuotaExceedAllowed, boolean
         .entrySet()
         .stream()
         .collect(Collectors.toMap(entry -> QuotaName.valueOf(entry.getKey()), Map.Entry::getValue));
-    return quotaManager.chargeAndRecommend(restRequest, requestCost, shouldCheckQuotaExceedAllowed, forceCharge);
+    QuotaAction quotaAction =
+        quotaManager.chargeAndRecommend(restRequest, requestCost, shouldCheckQuotaExceedAllowed, forceCharge);
+    // Note that we track quota usages even if QuotaMode is TRACKING, so although we return ALLOW, checkAndCharge still
+    // needs to happen
+    quotaAction = quotaManager.getQuotaMode() == QuotaMode.TRACKING ? QuotaAction.ALLOW : quotaAction;
```

Comment:
if we want to keep PostPorcessQuotaChargeCallback implementation, then we should probably move this logic to AmbryQuotaManager.

Reply:
+1

Reply:
I will refactor it as a cleanup effort. We will eventually get rid of PostProcessQuotaChargeCallback (and maybe the entire concept of callback class).